plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -85,25 +80,8 @@ class RsyncConf:
 
 @dataclass
 class EnvSetupConf:
-    pip_packages: Dict[str, str] = field(
-        default_factory=lambda: {
-            "omegaconf": omegaconf.__version__,
-            "hydra_core": hydra.__version__,
-            "ray": ray.__version__,
-            "cloudpickle": cloudpickle.__version__,
-            "pickle5": pkg_resources.get_distribution("pickle5").version,
-            "hydra_ray_launcher": pkg_resources.get_distribution(
-                "hydra_ray_launcher"
-            ).version,
-        }
-    )
-
-    commands: List[str] = field(
-        default_factory=lambda: [
-            "conda create -n hydra_${python_version:micro} python=${python_version:micro} -y",
-            "echo 'export PATH=\"$HOME/anaconda3/envs/hydra_${python_version:micro}/bin:$PATH\"' >> ~/.bashrc",
-        ]
-    )
```

Comment:
Can you explain this change?

Reply:
If a user wants this behavior, they can put it in the config file. If they don't want it, they have to go in and modify the plugin source. Why should the plugin be installing packages and modifying my `.bashrc` at runtime?

Reply:
Unless I'm mistaken, it's completely redundant with the [example config](https://hydra.cc/docs/plugins/ray_launcher) in the documentation.  As a user, it was very strange to remove the corresponding lines in the config and still have the launcher reinstalling packages.

Reply:
Commenting below, but @jieru-hu mayhave more context.

Installing things: The plugin is preparing your remove environment and ensuring it matches your local environment in terms of versions (having a version mismatch between the environment that launches the jobs and the environment that runs them is a great way to have the most fascinating bugs.
Also, even though you may choose to bake all the dependencies into your AMI or docker image, some users will not want to do that.

The change of the .bashrc indeed seems weird, I don't think we should be doing it.

You never have to change the plugin source to override configs. I suggest you go over the Structured Configs tutorial.

Reply:
Unless there's something I'm missing, if I have this in my config, that _should_ override the default EnvConfig, but that doesn't seem to be what actually happens. 
```
env_setup:
  pip_packages: {}
  commands: []
```

I think [this line](https://github.com/facebookresearch/hydra/blob/9e1039941ee2b714d043e15145fc358f4ab6e9b8/plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/_core_aws.py#L63) is to blame. Maybe that is what should be changed?

Reply:
Generally speaking, dictionaries are being merged. you can't delete things from dicts using merge.
Well, the pip packages there are required. If you really want to delete them,I guess we could support an explicit assignment of None:
```
pip_packages:
  hydra_core: null
  omegaconf: null
  ...
```
Supporting it would take a minor change to the code that generates the setup commands (filter anything that is None).

Can you explain why you don't think you need those packages in your remote environment?

Reply:
I have no doubt they're required -- but if you're running a  containerized app there's really no reason for them to be installed at runtime. I understand the versioning rationale... version discrepancy bugs suck. If you think keeping them there makes for a better experience for the average user I'm def not interested in dying on this hill. To me it seemed very strange that if I knew what I was doing I couldn't override the behavior without literally digging through the code and commenting the offending lines out.

Reply:
actually I just remembered why I did this. To test my changes I was installing the plugin from source in the container. I wanted to be sure it wasn't reinstalling from pypi.

Reply:
It's also possible (though maybe unlikely) that the user doesn't have pip in their container (maybe pip3, for example). In that case the head node setup fails.

Reply:
@omry So are we agreed that I make `EnvSetupConf.commands` an empty ListConfig and leave `EnvSetupConf.pip_packages` as is?

Reply:
In CI, the plugin is testing the proper code and not the pip package (in fact, our Amazon image does not even have access to the Internet for security reasons).
I am fine with adding support for removing mandatory pip packages by overriding them to null (which is the None in yaml).
Essentially, it should be as easy filtering any package with a None version in the code.

You should be able to override the commands to an empty list from your side. What I said about dictionaries does not apply to lists (Lists are always replacing the source list).

Reply:
It looks like if you don't prepend the conda path to `PATH` in `.bashrc` then the head node setup fails because the `PyYAML` install fails
```
Collecting PyYAML>=5.1.*
  Downloading https://files.pythonhosted.org/packages/64/c2/b80047c7ac2478f9501676c988a5411ed5572f35d1beff9cae07d321512c/PyYAML-5.3.1.tar.gz (269kB)
     |████████████████████████████████| 276kB 7.8MB/s
Building wheels for collected packages: PyYAML
  Building wheel for PyYAML (setup.py) ... done
  Created wheel for PyYAML: filename=PyYAML-5.3.1-cp36-cp36m-linux_x86_64.whl size=44621 sha256=3834687cdd8b382c59c8a02769f50093619ad10ecc392500ff6fbb274524eed2
  Stored in directory: /home/ubuntu/.cache/pip/wheels/a7/c1/ea/cf5bd31012e735dc1dfea3131a2d5eae7978b251083d6247bd
Successfully built PyYAML
Installing collected packages: typing-extensions, dataclasses, PyYAML, omegaconf
  Found existing installation: PyYAML 3.12
ERROR: Cannot uninstall 'PyYAML'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
```